Userland/Applications/Escalator/EscalatorWindow.cpp

```diff
@@ -0,0 +1,140 @@
+{
+    auto app_icon = find_icon();
+
+    set_title("Run as Root");
+    set_icon(app_icon.bitmap_for_size(16));
+    resize(345, 100);
+    set_resizable(false);
+    set_minimizable(false);
+
+    auto& main_widget = set_main_widget<GUI::Widget>();
+    main_widget.load_from_gml(escalator_gml);
+
+    RefPtr<GUI::Label> app_label = *main_widget.find_descendant_of_type_named<GUI::Label>("description");
+
+    String prompt;
+    if (description.is_empty())
+        prompt = String::formatted("{} requires root access. Please enter password for user \"{}\".", command[0], m_current_user.username());
+    else
+        prompt = description;
+
+    app_label->set_text(prompt);
+
+    m_icon_image_widget = *main_widget.find_descendant_of_type_named<GUI::ImageWidget>("icon");
+    m_icon_image_widget->set_bitmap(app_icon.bitmap_for_size(32));
+
+    m_ok_button = *main_widget.find_descendant_of_type_named<GUI::DialogButton>("ok_button");
+    m_ok_button->on_click = [this](auto) {
+        auto result = check_password();
+        if (result.is_error()) {
+            hide();
+            GUI::MessageBox::show_error(this, String::formatted("Failed to execute command: {}", result.error()));
+            close();
+        }
+    };
+    m_ok_button->set_default(true);
+
+    m_cancel_button = *main_widget.find_descendant_of_type_named<GUI::DialogButton>("cancel_button");
+    m_cancel_button->on_click = [this](auto) {
+        close();
+    };
+
+    m_password_input = *main_widget.find_descendant_of_type_named<GUI::PasswordBox>("password");
+}
+
+GUI::Icon EscalatorWindow::find_icon()
+{
+    String icon_path = m_command[0];
+
+    // Path to executable directly provided.
+    if (icon_path.find('/').has_value())
+        return GUI::FileIconProvider::icon_for_executable(icon_path);
+
+    // Find the application executable on the user's PATH.
+    // If PATH is absent from the environ for whatever reason, fallback on /bin.
+    String path = "/bin";
+    for (size_t i = 0; environ[i]; ++i) {
+        StringView environ_setting { environ[i], strlen(environ[i]) };
+        auto split = environ_setting.split_view('=');
+
+        if (split.size() == 2 && split[0] == "PATH" && split[1].length()) {
+            path = split[1];
+            break;
+        }
+    }
+
+    Vector<StringView> paths = path.split_view(':');
+    for (auto const& path_entry : paths) {
+        String full_path = String::formatted("{}/{}", path_entry, m_command[0]);
+        auto file_or_error = Core::MappedFile::map(full_path);
+        if (!file_or_error.is_error())
+            return GUI::FileIconProvider::icon_for_executable(full_path);
+    }
+
+    return GUI::Icon::default_icon("app-escalator"sv);
+}
+
+ErrorOr<void> EscalatorWindow::check_password()
+{
+    String password = m_password_input->text();
+    if (password.is_empty()) {
+        hide();
+        GUI::MessageBox::show_error(this, "Please enter a password."sv);
+        show();
+        return {};
```

Comment:
We can probably just leave the background window visible.

Reply:
So if we keep the dialogue open, with the current MessageBox API it looks like we'll end up with errors that look like this:

![image](https://user-images.githubusercontent.com/4776559/194725846-567544eb-e01b-4ad4-8427-3cf4c1d628e4.png)

MessageBox will center it on the dialog itself rather than center it on-screen. If we omit `this` in the argument, the box will center on desktop, but we lose modal functionality. RunWindow, the application I modeled much of the behaviour of Escalator after, does something similar where it hides the main window to display error dialogues in the center of the screen. 

Removing the hides and shows is trivial but, unless there's a way of manually positioning the error dialogue, it looks like it will come with that as a side effect.

Reply:
Following discussion in discord, will go with option where error dialogue is centered on Escalator application.